src/category_theory/action.lean

```diff
@@ -0,0 +1,94 @@
+/-
+Copyright (c) 2020 David Wärn. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Wärn
+-/
+
+import category_theory.groupoid
+import category_theory.endomorphism
+import category_theory.elements
+import group_theory.group_action
+import category_theory.single_obj
+
+/-!
+# Actions as functors and as categories
+
+From a multiplicative action M ↻ X, we can construct a functor from M to the category of
+types, mapping the single object of M to X and an element `m : M` to map `X → X` given by
+multiplication by `m`.
+  This functor induces a category structure on X -- a special case of the category of elements.
+A morphism `x → y` in this category is simply a scalar `m : M` such that `m • x = y`. In the case
+where M is a group, this category is a groupoid -- the `action groupoid'.
+-/
+
+open mul_action
+namespace category_theory
+
+universes u
+
+variables (M : Type*) [monoid M] (X : Type u) [𝒜 : mul_action M X]
+include 𝒜
+
+/-- A multiplicative action M ↻ X viewed as a functor in the natural way. -/
```

Comment:
Let's say something more concrete than "in the natural way". The module doc sentence is fine.